dlup/annotations.py

```diff
@@ -495,30 +566,69 @@ def from_halo_xml(cls, halo_xml: PathLike, scaling: float | None = None) -> WsiA
+                _cls = AnnotationClass(label=layer.name, a_cls=AnnotationType.POLYGON)
+                for shapely_polygon in shapely_multipolygon.geoms:
+                    curr_polygon = rescale_geometry(Polygon(shapely_polygon, a_cls=_cls), scaling=scaling)
+                    output[layer.name].append(Polygon(curr_polygon, a_cls=_cls))
 
         annotations: list[WsiSingleLabelAnnotation] = []
         for label in output:
             annotations.append(
-                WsiSingleLabelAnnotation(label=label, type=AnnotationType.POLYGON, coordinates=output[label])
+                WsiSingleLabelAnnotation(
+                    a_cls=AnnotationClass(label=label, a_cls=AnnotationType.POLYGON), coordinates=output[label]
+                )
             )
 
         return cls(annotations)
 
-    def __getitem__(self, label: str) -> WsiSingleLabelAnnotation:
-        return self._annotations[label]
+    @classmethod
+    def from_darwin_json(cls, darwin_json: PathLike, scaling: float | None = None) -> WsiAnnotations:
+        if not DARWIN_SDK_AVAILABLE:
+            raise RuntimeError("`darwin` is not available. Install using `python -m pip install darwin-py`.")
+
+        annotations = defaultdict(list)
+        _scaling = 1.0 if not scaling else scaling
+
+        darwin_an = darwin.utils.parse_darwin_json(pathlib.Path(darwin_json), None)
+
+        for curr_annotation in darwin_an.annotations:
+            name = curr_annotation.annotation_class.name
+            annotation_type = _v7_annotation_type_to_dlup_annotation_type(
+                curr_annotation.annotation_class.annotation_type
+            )
+            key = AnnotationClass(label=name, a_cls=annotation_type)
+            curr_data = curr_annotation.data
+
+            _cls = AnnotationClass(label=name, a_cls=annotation_type)
+            if annotation_type == AnnotationType.POINT:
+                curr_point = Point((curr_data["x"], curr_data["y"]), a_cls=_cls)
+                curr_point = rescale_geometry(curr_point, scaling=scaling)
+                annotations[key].append(curr_point)
+            elif annotation_type == AnnotationType.POLYGON:
+                curr_polygon = Polygon([(_["x"], _["y"]) for _ in curr_data["path"]], a_cls=_cls)
+                curr_polygon = rescale_geometry(curr_polygon, scaling=scaling)
+                annotations[key].append(Polygon(curr_polygon, a_cls=_cls))
+            elif annotation_type == AnnotationType.BOX:
+                x, y, h, w = curr_data.values()
+                curr_polygon = shapely.geometry.box(x, y, x + w, y + h)
+                curr_polygon = rescale_geometry(curr_polygon, scaling=scaling)
+                annotations[key].append(Polygon(curr_polygon, a_cls=_cls))
+            else:
+                ValueError(f"Annotation type {annotation_type} is not supported.")
+
+        # Now we can make WsiSingleLabel annotations
+        output = []
+        for an_cls in annotations:
```

Comment:
```python
        for an_cls, coordinates in annotations.items():
            output.append(WsiSingleLabelAnnotation(a_cls=an_cls, coordinates=coordinates))
        return cls(output)
```

Reply:
Just a very small change to make it a bit more explicit that we get the "coordinates"

Reply:
Done